modules/gcloud/src/main/java/org/testcontainers/containers/DatastoreEmulatorContainer.java

```diff
@@ -21,13 +21,32 @@ public class DatastoreEmulatorContainer extends GenericContainer<DatastoreEmulat
 
     private static final int HTTP_PORT = 8081;
 
+    private String flags;
+
+    public DatastoreEmulatorContainer(final String image) {
+        this(DockerImageName.parse(image));
+    }
+
     public DatastoreEmulatorContainer(final DockerImageName dockerImageName) {
         super(dockerImageName);
         dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
 
         withExposedPorts(HTTP_PORT);
         setWaitStrategy(Wait.forHttp("/").forStatusCode(200));
-        withCommand("/bin/sh", "-c", CMD);
+    }
+
+    @Override
+    protected void configure() {
+        String command = CMD;
+        if (this.flags != null && !this.flags.isEmpty()) {
+            command += " " + this.flags;
```

Comment:
Don't we have to join all flags separated by spaces?

Reply:
I think the trick is the next line `withCommand("/bin/sh", "-c", command);` due to the initial command is `gcloud beta emulators datastore start --project test-project --host-port 0.0.0.0:8081` we are just adding a space for the upcoming flags.